pages/Navbar/Navbar.js

```diff
@@ -0,0 +1,109 @@
+import { useSession, signIn, signOut } from "next-auth/react";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+
+export default function Navbar() {
+  const { data: session, status } = useSession();
+  const router = useRouter();
+
+  if (status === "loading") {
+    return <LoadingText>Loading...</LoadingText>;
+  }
+
+  return (
+    <NavbarContainer>
+      <UserSection>
+        {session ? (
+          <>
+            {session.user.image ? (
+              <Avatar src={session.user.image} alt={session.user.name} />
+            ) : (
+              <InitialAvatar>
+                {session.user.name.charAt(0)} {/* Display first letter */}
+              </InitialAvatar>
+            )}
+            <FullName>{session.user.name}</FullName>
+            <SignInButton onClick={() => signOut()}>Sign Out</SignInButton>
+          </>
+        ) : (
+          <SignInButton
+            onClick={() =>
+              signIn("credentials", { callbackUrl: router.asPath })
+            }
+          >
+            Sign In
+          </SignInButton>
+        )}
+      </UserSection>
+    </NavbarContainer>
+  );
+}
+
+const NavbarContainer = styled.div`
```

Comment:
It's called `NavbarContainer` but it is not a `nav` element. This might be confusing.

In my opinion, the best would be to rename this entire component. This is not really a navigation - this component displays user data. I suggest to rename it to `UserMenu`.

The container element could be a `section`.

```suggestion
const UserMenuSection = styled.section`
```